packages/app/demo/FrameView.tsx

```diff
@@ -100,20 +100,26 @@ export const FramePlayer: React.FC<{ player: Player }> = ({ player }) => {
 }
 
 export const TrailView: React.FC = () => {
+  const frame = useMatch((state) => state.frame)
   const round = useMatch((state) => state.round)
   const ref = React.useRef<HTMLCanvasElement>(null)
   React.useEffect(() => {
     const context = ref.current?.getContext?.("2d")
     if (context) {
       context.clearRect(0, 0, 1024, 1024)
-      round?.Frames.forEach((e) => {
-        for (const player of e.Players) {
-          context.fillStyle = teamColor(player.Team)
-          context.fillRect(player.X, player.Y, 1, 1)
-        }
-      })
+      if (frame) {
+        const index = round?.Frames.indexOf(frame)
+        const futureFrames = round?.Frames.slice(index)
+
+        futureFrames?.forEach((e) => {
+          for (const player of e.Players) {
+            context.fillStyle = teamColor(player.Team)
+            context.fillRect(player.X, player.Y, 1, 1)
+          }
+        })
+      }
     }
-  }, [round])
+  }, [frame])
```

Comment:
this is effectively correct but, it will be more safe to keep `round` in the deps.

```suggestion
  }, [frame, round])
```